### ISSUE TITLE
fast_update not work with highlights/stories when in commit-mode 

### ISSUE THREAD

**Issue by pjw91** [instaloader/instaloader#257]
**Describe the bug**
When downloading highlights/stories with the commit-mode and fast-update are on,
the fast-update has no effect.

**To Reproduce**
instaloader -F --highlights SOME_PROFILE --commit-mode --no-posts

**Expected behavior**
Skip those downloaded, as if commit-mode off.

**Instaloader version**
4.2.3

**Additional context**
The reason is that `self._committed` is never set in `download_highlights` and `download_stories`, and never reset to None after a post is processed (either downloaded or skipped).

That means:
1. `_committed` is always `None`, which effectively disables commit-mode
1. After processing a post, the flag is set to True/False, no matter what actually status of the next highlights/stories is.

After a quick review, it seems that add a line of code `self._committed = self.check_if_committed(filename)` at the top of `download_pic` fix this issue.
However, I think it should be add to `download_stories/highlights` or somewhere else.

**Comment by aandergr** [comment]
Thanks for reporting this and sharing your insights!

> After a quick review, it seems that add a line of code `self._committed = self.check_if_commited(filename)` at the top of `download_pic` fix this issue.

That would check_if_committed for each item in a sidecar, though one check would be sufficient.

> However, I think it should be add to `download_stories/highlights` or somewhere else.

Or `download_storyitem`, which is used both for downloading stories and highlights.

Also, I think passing this _committed state through a class variable is unclean, error-prone and blocking thread-parallelization - we may have merged the PR (#139) too early and probably should reimplement the commit mode. @sushilicious any ideas/thoughts about this?

**Comment by davispuh** [comment]
This looks fixed/resolved to me, no ?

**Comment by aandergr** [comment]
> This looks fixed/resolved to me, no ?

No, there hasn't been any work on this.

**Comment by sushilicious** [comment]
Yeah sorry I've never used the stories or highlights feature, so I didn't think to implement the commit mode for that. As is mentioned in the first post, adding the check for committed in `download_stories/highlights` is probably the right thing to do.

@aandergr, sounds totally reasonable to change the _committed state to something else. The patch was mostly just a quick and dirty thing that I wrote up since the lack of a commit state was bothering me. If someone else plans to properly implement this feature, I also recommend using something other than the json metadata file as the flag for "already committed". That file can get quite large, so a different indicator might make more sense. For example, saving currently downloading pictures/videos to a temp file and renaming it after it is done, so uncommitted pics/videos do not enter your download folder.

**Comment by aandergr** [comment]
This has been fixed by #697.

> Rather than checking the json file to make sure posts have been successfully downloaded, data is stored in a temporary file which is renamed when downloading has finished.

It is merged into `upcoming/v4.5`, which can be tried out with:

```
pip3 install --upgrade git+https://github.com/instaloader/instaloader@upcoming/v4.5
```
